execution_engine_testing/test_support/src/chainspec_config.rs

```diff
@@ -0,0 +1,190 @@
+            toml::from_slice(&bytes).map_err(Error::FailedToParseChainspec)?;
+        Ok(chainspec_config)
+    }
+
+    pub(crate) fn create_genesis_request_from_chainspec<P: AsRef<Path>>(
+        filename: P,
+        genesis_accounts: Vec<GenesisAccount>,
+        protocol_version: ProtocolVersion,
+    ) -> Result<RunGenesisRequest, Error> {
+        let path = filename.as_ref();
+        let bytes = fs::read(path).map_err(|error| Error::FailedToLoadChainspec {
+            path: path.into(),
+            error,
+        })?;
+        let chainspec_config: ChainspecConfig =
+            toml::from_slice(&bytes).map_err(Error::FailedToParseChainspec)?;
+        let locked_funds_period_millis =
+            humantime::parse_duration(&*chainspec_config.core_config.locked_funds_period)
+                .map_err(|_| Error::CouldNotCreateGenesisRequest)?
+                .as_millis() as u64;
+        let exec_config = ExecConfig::new(
+            genesis_accounts,
+            chainspec_config.wasm_config,
+            chainspec_config.system_costs_config,
+            chainspec_config.core_config.validator_slots,
+            chainspec_config.core_config.auction_delay,
+            locked_funds_period_millis,
+            chainspec_config.core_config.round_seigniorage_rate,
+            chainspec_config.core_config.unbonding_delay,
+            DEFAULT_GENESIS_TIMESTAMP_MILLIS,
+        );
+        Ok(RunGenesisRequest::new(
+            *DEFAULT_GENESIS_CONFIG_HASH,
+            protocol_version,
+            exec_config,
+            DEFAULT_CHAINSPEC_REGISTRY.clone(),
+        ))
+    }
+
+    /// Create a `RunGenesisRequest` using values from the production `chainspec.toml`.
+    pub fn create_genesis_request_from_production_chainspec(
+        genesis_accounts: Vec<GenesisAccount>,
+        protocol_version: ProtocolVersion,
+    ) -> Result<RunGenesisRequest, Error> {
+        Self::create_genesis_request_from_chainspec(
+            &*PRODUCTION_PATH,
+            genesis_accounts,
+            protocol_version,
+        )
+    }
+}
+
+impl TryFrom<ChainspecConfig> for ExecConfig {
+    type Error = Error;
+
+    fn try_from(chainspec_config: ChainspecConfig) -> Result<Self, Self::Error> {
+        let locked_funds_period_millis =
+            humantime::parse_duration(&*chainspec_config.core_config.locked_funds_period)
+                .map_err(|_| Error::CouldNotCreateExecConfig)?
+                .as_millis() as u64;
+        Ok(ExecConfig::new(
+            DEFAULT_ACCOUNTS.clone(),
+            chainspec_config.wasm_config,
+            chainspec_config.system_costs_config,
+            chainspec_config.core_config.validator_slots,
+            chainspec_config.core_config.auction_delay,
+            locked_funds_period_millis,
+            chainspec_config.core_config.round_seigniorage_rate,
+            chainspec_config.core_config.unbonding_delay,
+            DEFAULT_GENESIS_TIMESTAMP_MILLIS,
+        ))
+    }
+}
```

Comment:
These types are part of the `node` crate and have been copied over. I'm wondering whether it wouldn't make sense to move them now to `casper-types`. Both `execution-engine-testing` and `node` depend on that crate (`casper-types`) so the code would be available to both. There's another advantage - since this approach (of allowing people to use a chainspec for their tests) requires this code to be in sync with latest production chainspec, it does make sense to share the type definitions. Otherwise, users will have runtime errors if the code here doesn't get updated with the chainspec. Sharing it with node will make sure that it never happens. And there won't be unnecessary code duplication.